akka-actor-typed-tests/src/test/scala/akka/actor/typed/DispatcherSelectorSpec.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed
+
+import akka.actor.ActorCell
+import akka.actor.testkit.typed.scaladsl.{ LogCapturing, ScalaTestWithActorTestKit }
+import akka.actor.typed.internal.adapter.ActorContextAdapter
+import akka.actor.typed.scaladsl.AskPattern._
+import akka.actor.typed.scaladsl.Behaviors
+import akka.dispatch._
+import org.scalatest.wordspec.AnyWordSpecLike
+
+class DispatcherSelectorSpec extends ScalaTestWithActorTestKit("""
+    specific-dispatcher {
+      type = Dispatcher
+      executor = "thread-pool-executor"
+    }
+    specific-mailbox {
+      mailbox-type = "akka.dispatch.NonBlockingBoundedMailbox"
+      mailbox-capacity = 4
+    }
+    specific-dispatcher-with-mailbox {
+      type = Dispatcher
+      executor = "thread-pool-executor"
+      mailbox-type = "akka.dispatch.NonBlockingBoundedMailbox"
+      mailbox-capacity = 4
+    }
+  """) with AnyWordSpecLike with LogCapturing {
+
+  case class WhatsYourDispatcherAndMailbox(replyTo: ActorRef[(String, MessageQueue)])
+
+  private def behavior: Behavior[WhatsYourDispatcherAndMailbox] =
+    Behaviors.setup { context =>
+      Behaviors.receiveMessage[WhatsYourDispatcherAndMailbox] {
+        case WhatsYourDispatcherAndMailbox(replyTo) =>
+          val result = context match {
+            case adapter: ActorContextAdapter[_] =>
+              adapter.classicActorContext match {
+                case cell: ActorCell =>
+                  (cell.dispatcher.id, cell.mailbox.messageQueue)
+                case unexpected => throw new RuntimeException(s"Unexpected: $unexpected")
+              }
+            case unexpected => throw new RuntimeException(s"Unexpected: $unexpected")
+          }
+
+          replyTo ! result
+          Behaviors.stopped
+      }
+    }
+
+  "The Dispatcher selectors" must {
+
+    "default to unbounded" in {
+      val actor = spawn(behavior)
+      val (_, mailbox) = actor.ask(WhatsYourDispatcherAndMailbox.apply).futureValue
+      mailbox shouldBe a[UnboundedMessageQueueSemantics]
```

Comment:
should this be even more specific?
```
mailbox shouldBe a[NodeMessageQueue]
```

Reply:
Thanks for the suggestion. However, when I received the CI complaint, I noticed that there is already a DispatcherSelectorSpec in akka.actor.typed.scaladsl. So, I want to figure out how to merge them. I will fix those issues and commit later.